### ISSUE TITLE
&amp; TO & amp;

### ISSUE THREAD

**Issue by ernes32** [beautify-web/js-beautify#880]
For example if you try to beautify this line:
`for(i=0;i<A.sampleRate/2;i++)cB[i]=(((i*13+(((i>>3)&amp;1221)+133)*(i>>2))&amp;255)/128-.4)/(i*.00002+1)*(Math.cos(i*i*.0000003+i*0.40321+j)%0.412)*12;`

you will get an error due to unresolved new variable "amp;"

btw. this was taken from @notch http://doodle.notch.net/unmandelboxing/withsound.html

**Comment by bitwiseman** [comment]
That's because `&amp;` is not javascript, it is html/xml encoding of javascript. The code in question was in a `script` tag on an html page, right?

**Comment by ernes32** [comment]
Yes true the code was in a `script` tag. Sorry for bothering you with this.
On the other hand if you can make lexer recognize this case it would be nice.
Thank you.

**Comment by bitwiseman** [comment]
No apology needed.  I was just making sure I understood the context of the input since you only showed javascript. 

I totally see the value.

**Comment by bitwiseman** [comment]
Related #869

**Comment by BryanH** [comment]
That is invalid JavaScript anyway, so it may not be an issue unless there's some other reason for escaping the ampersands.

The only time you'd want to escape an ampersand (or other HTML element), is if you're writing a string to the DOM.

I put together a JS Fiddle to demonstrate this (expires Nov 2026): [Escapes don't work in JS code](https://jsfiddle.net/SCraquil/ac3vr8o7/)
